Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -551,19 +555,31 @@ public override void Connect()
 
                     if (!_client.Connected) throw new Exception("InteractiveBrokersBrokerage.Connect(): Connection returned but was not in connected state.");
 
-                    if (!DownloadAccount())
+                    if (IsFinancialAdvisor)
                     {
-                        Log.Trace("InteractiveBrokersBrokerage.Connect(): DownloadAccount failed. Operation took longer than 15 seconds.");
-
-                        Disconnect();
-
-                        if (attempt++ < maxAttempts)
+                        if (!DownloadFinancialAdvisorAccount(_account))
                         {
-                            Thread.Sleep(1000);
-                            continue;
+                            Disconnect();
+
+                            throw new TimeoutException("InteractiveBrokersBrokerage.Connect(): DownloadFinancialAdvisorConfiguration failed.");
```

Comment:
Do we want similar re-try logic here as below?

```c#
if (attempt++ < maxAttempts)
{
    Thread.Sleep(1000);
    continue;
 }
```

Reply:
Fixed